src/cmd/services/m3coordinator/downsample/downsampler.go

```diff
@@ -36,11 +36,17 @@ type Downsampler interface {
 // appender, only valid to use with a single caller at a time.
 type MetricsAppender interface {
 	AddTag(name, value []byte)
-	SamplesAppender(opts SampleAppenderOptions) (SamplesAppender, error)
+	SamplesAppender(opts SampleAppenderOptions) (SamplesAppenderResult, error)
 	Reset()
 	Finalize()
 }
 
+// SamplesAppenderResult is the result from a SamplesAppender call.
+type SamplesAppenderResult struct {
+	SamplesAppender     SamplesAppender
+	IsDropPolicyApplied bool
```

Comment:
nit; Would this be better if it was a method on SamplesAppender?

Reply:
So that;s what I had earlier. But it becomes odd as samplesAppender returned is a multi-samples appender, so it does not really fit the paradigm well. Also the face that drop policy was applied is separate from the appender itself.